### ISSUE TITLE
A member of `Function.prototype` can accessible from a type which is intersected with an arrow function, type argument, and nominal branding members but it's not assignable to the `Function` typed argument

### ISSUE THREAD

**Issue by acid-chicken** [microsoft/TypeScript#38497]
**TypeScript Version:**  3.9.2

**Search Terms:**

**Expected behavior:**

TS2345 should NOT occur (It did not happen in ver 3.8.3).

**Actual outcomes:**

TS2345 is raised as follows:

> Argument of type 'T & { readonly [fooBrand]?: undefined; readonly [barBrand]?: undefined; } & Callable' is not assignable to parameter of type 'Function'. Type 'T & { readonly [fooBrand]?: undefined; readonly [barBrand]?: undefined; } & Callable' is missing the following properties from type 'Function': apply, call, bind, toString, and 6 more.

**Related Issues:**

#36696?

**Code**
```ts
declare const fooBrand: unique symbol
declare const barBrand: unique symbol

type Callable = (...args: readonly any[]) => any
type CallableXPlugin = {
    readonly [fooBrand]?: never
    readonly [barBrand]?: never
} & Callable
type FooXPlugin = {
    readonly [fooBrand]: 0
    readonly [barBrand]?: never
}
type BarXPlugin = {
    readonly [fooBrand]?: never
    readonly [barBrand]: 0
}
type XPlugin = CallableXPlugin | FooXPlugin | BarXPlugin

function isCallable(x: XPlugin): x is CallableXPlugin {
    return typeof x == 'function'
}
function yo<T extends XPlugin>(x: T) {
    if (isCallable(x)) {
        x.apply(null) // no error
        Reflect.apply(x, null, [] as const) // error!
    }
}

```

<details><summary><b>Output</b></summary>

```ts
"use strict";
function isCallable(x) {
    return typeof x == 'function';
}
function yo(x) {
    if (isCallable(x)) {
        x.apply(null); // no error
        Reflect.apply(x, null, []); // error!
    }
}

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "alwaysStrict": true,
    "esModuleInterop": true,
    "declaration": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "moduleResolution": 2,
    "target": "ES2017",
    "jsx": "React",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/v2/en/play?ts=3.9.2#code/CYUwxgNghgTiAEYD2A7AzgF3gMyUgQjFCsAFzwCuKAlgI4UJoCeAtgEZIQBQoksCydFjaxCxMpRr1GrDty4YmABwQBhKBGhsICALzwAFADoTsAOZpycKMFQQm8YkwDaAXQCU8XQD5HKJgrKahpaOgAaAAoQFGbUKF7wAN5c8Knw1rYo9vDOuAREJK4A-OQoIABuIDApaRl2Ds4iMGKFJfBlldUAvvAAZPDqmlDaIIEq8ABieJHRsfH6yWnpIDb1OXktwK7kAAw1qXVZDU2bxaUVVVxdYwj4sDMxcQmLtSuZ2bl4p20dl0uHHxOBS2uyuN3gDzmCUGoRAkKeAB9JtMoo94ki7jB4SguFxsFQwBhqKh4NQ0DDhjoDAAPcjY9zkamktADEKUuGoqEvA4gDAUGDxRQqJDYeBM3T6ADk+JQhOJKElYJlcpJTCQAB4ACrwEDUjAgEgs7HeGnkTWebmk0UGMkUkY09wW-ZLMVGKBKJT2AwoCiaTwAen97SQOpgMCQ1RdaQASiBsDpCW6PV7qQAadq+iDptyOFmCTABoNVcMwACEzuu1yAA)

**Comment by acid-chicken** [comment]
I'm sorry, it seems that this is a duplicate of #38549.
